03-rnaseq/pathway-analysis_rnaseq_03_gsva.Rmd

```diff
@@ -0,0 +1,664 @@
+
+The output is a gene set by sample matrix of GSVA scores.
+
+### Perform GSVA
+
+Let's perform GSVA using the `gsva()` function.
+See `?gsva` for more options.
+
+```{r}
+gsva_results <- gsva(
+  filtered_mapped_matrix,
+  hallmarks_list,
+  method = "gsva",
+  # Appropriate for our normalized counts data
+  kcdf = "Gaussian",
+  # Minimum gene set size
+  min.sz = 15,
+  # Maximum gene set size
+  max.sz = 500,
+  # Compute Gaussian-distributed scores
+  mx.diff = TRUE,
+  # Don't print out the progress bar
+  verbose = FALSE
+)
+```
+
+Note that the `gsva()` function documentation says we can use `kcdf = "Gaussian"` if we have expression values that are continuous such as log-CPMs, log-RPKMs or log-TPMs, but we would use `kcdf = "Poisson"` on integer counts.
+Our `vst()` transformed data is log, so `Gaussian` works for us. 
+
+Let's explore what the output of `gsva()` looks like.
+
+```{r}
+head(gsva_results[, 1:10])
+```
+
+## Write results to file
+
+Let's write all of our GSVA results to file.
+
+```{r}
+gsva_results %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("pathway") %>%
+  readr::write_tsv(file.path(
+    results_dir,
+    "SRP140558_gsva_results.tsv"
+  ))
+```
+
+**Review**
+
+## Visualizing Results with a heatmap
+
+Let's make a heatmap for our pathways! 
+
+## Neaten up our metadata labels
+
+We will want our heatmap to include our metadata labels, but unfortunately some of the metadata for this dataset are not set up into separate, neat columns. 
+No worries, this is something we will accomplish ourselves by creating a new data frame with the labels we want shown in the heatmap. 
+
+For this study, PBMCs were collected at two time points: during the patients' first, acute bronchiolitis visit (abbreviated "AV") and their recovery visit, (called post-convalescence and abbreviated "CV").
+
+We can create a new variable, `time_point`, that states this info more clearly. 
+This new `time_point` variable will have two labels: `acute illness` and `recovering` based on the `AV` or `CV` coding located in the `refinebio_title` string variable. 
+
+```{r}
+annot_df <- metadata %>%
+  # We need the sample IDs and the main column that contains the metadata info
+  dplyr::select(
+    refinebio_accession_code,
+    refinebio_title
+  ) %>%
+  # Create our `time_point` variable based on `refinebio_title`
+  dplyr::mutate(
+    time_point = dplyr::case_when(
+      # Create our new variable based on refinebio_title containing AV/CV
+      stringr::str_detect(refinebio_title, "_AV_") ~ "acute illness",
+      stringr::str_detect(refinebio_title, "_CV_") ~ "recovering"
+    ),
+    # It's easier for future items if this is already set up as a factor
+    time_point = as.factor(time_point)
+  ) %>%
+  # We don't need the older version of the variable anymore
+  dplyr::select(-refinebio_title)
```

Comment:
Do we need these steps?

Reply:
Which part? The refinebio_title has unintelligible items like: `"AVB_111_CV_PBMC"` and there's not a metadata column that has this information separated out so we need to do some rearrangement. 

But any columns left in this data frame are automatically added to the plot by pheatmap, and I don't find that useful so that's why I'm dropping this column after we get the info from it. (thought about using a transmute but it ended up more confusing).

Reply:
> But any columns left in this data frame are automatically added to the plot by pheatmap, and I don't find that useful so that's why I'm dropping this column after we get the info from it. (thought about using a transmute but it ended up more confusing).

Okay, makes sense. But the factor conversion?

Reply:
Oh factor conversion can probably be dropped now. 🎉